### ISSUE TITLE
Highlight and indentation when method def starts with "private"

### ISSUE THREAD

**Issue by rmolinari** [zenspider/enhanced-ruby-mode#108]
Sometimes a method is defined as private like this:
```
class Foo
  private def bar
            x = 1
            x
          end
end
```

But enh-ruby is indenting oddly, like that. The method body is indented against the `def` while I expect it to indent it against `private` like this:
```
class Foo
  private def bar
    x = 1
    x
  end
end
```
"Regular" ruby mode indents it like that.

Also, `private` does not get any syntax highlighting.

**Comment by rmolinari** [comment]
![enh_ruby_indent](https://user-images.githubusercontent.com/7924302/27148247-9764792a-510d-11e7-906e-475adc06f2dd.jpg)

**Comment by rmolinari** [comment]
I have Emacs v 25.2.1. 

Here is my enh-ruby configuration in init.el
```
(use-package enh-ruby-mode
  :ensure t
  
  :mode
  (("Capfile" . enh-ruby-mode)
   ("Gemfile\\'" . enh-ruby-mode)
   ("Rakefile" . enh-ruby-mode)
   ("\\.rb" . enh-ruby-mode)
   ("\\.ru" . enh-ruby-mode))

  :init
  (rename-modeline "enh-ruby-mode" enh-ruby-mode "Ruby")

  :config
  (use-package smartparens-ruby :ensure smartparens)
  (use-package inf-ruby)
  
  (setq enh-ruby-add-encoding-comment-on-save nil
        enh-ruby-deep-indent-paren nil
        enh-ruby-hanging-brace-indent-level 2
        ruby-insert-encoding-magic-comment nil
        enh-ruby-program "/usr/local/opt/ruby/bin/ruby")

  (add-to-list 'interpreter-mode-alist '("ruby" . enh-ruby-mode))

  (add-hook 'enh-ruby-mode-hook 'inf-ruby-minor-mode))
```

I use smartparens and smartparens-ruby.

**Comment by rmolinari** [comment]
Oh. I see that this is the same issue as #67.

It looks like there is not a desire to change this behavior.

**Comment by zenspider** [comment]
This should be fixed by @knu's #110.

**Comment by knu** [comment]
The `private` keyword not getting highlighted is a separate issue.  I've filed #114 which addresses it.
